packages/payments-plugin/src/mollie/mollie.helpers.ts

```diff
@@ -0,0 +1,79 @@
+import { CreateParameters } from '@mollie/api-client/dist/types/src/binders/orders/parameters';
+import { Amount } from '@mollie/api-client/dist/types/src/data/global';
+import { OrderAddress as MollieOrderAddress } from '@mollie/api-client/dist/types/src/data/orders/data';
+import { Customer, Order } from '@vendure/core';
+
+import { OrderAddress } from './graphql/generated-shop-types';
+
+/**
+ * Check if given address has mandatory fields for Mollie Order and map to a MollieOrderAddress.
+ * Returns undefined if address doesn't have all mandatory fields
+ */
+export function toMollieAddress(address: OrderAddress, customer: Customer): MollieOrderAddress | undefined {
+    if (address.city && address.countryCode && address.streetLine1 && address.postalCode) {
+        return {
+            streetAndNumber: `${address.streetLine1} ${address.streetLine2 || ''}`,
+            postalCode: address.postalCode,
+            city: address.city,
+            country: address.countryCode.toUpperCase(),
+            givenName: customer.firstName,
+            familyName: customer.lastName,
+            email: customer.emailAddress,
+        };
+    }
+}
+
+/**
+ * Map all order and shipping lines to a single array of Mollie order lines
+ */
+export function toMollieOrderLines(order: Order): CreateParameters['lines'] {
+    const lines: CreateParameters['lines'] = order.lines.map(line => ({
+        name: line.productVariant.name,
+        quantity: line.quantity,
+        unitPrice: toAmount(line.proratedUnitPriceWithTax, order.currencyCode), // totalAmount has to match unitPrice * quantity
+        totalAmount: toAmount(line.proratedLinePriceWithTax, order.currencyCode),
+        vatRate: String(line.taxRate),
+        vatAmount: toAmount(line.lineTax, order.currencyCode),
+    }));
+    // Add shippingLines
+    lines.push(...order.shippingLines.map(line => ({
+        name: line.shippingMethod?.name || 'Shipping',
+        quantity: 1,
+        unitPrice: toAmount(line.priceWithTax, order.currencyCode),
+        totalAmount: toAmount(line.discountedPriceWithTax, order.currencyCode),
+        vatRate: String(line.taxRate),
+        vatAmount: toAmount(line.discountedPriceWithTax - line.discountedPrice, order.currencyCode),
+    })));
+    return lines;
+}
+
+/**
+ * Stringify and fixed decimals
+ * @param value
+ */
+export function toAmount(value: number, orderCurrency: string): Amount {
+    return {
+        value: (value / 100).toFixed(2),
+        currency: orderCurrency,
+    };
+}
+
+/**
+ * Lookup one of Mollies allowed locales based on an orders countrycode or channel default.
+ * If both lookups fail, resolve to en_US to prevent payment failure
```

Comment:
What does `locale` signify in Mollie? Is it only for localization? If so this is fine.

Reply:
Yep, only for showing the customers language during the hosted checkout. It is a mandatory field for some reason.

> Allows you to preset the language to be used in the hosted payment pages shown to the consumer.